File: ComponentImpl/src/main/java/com/xiaojinzi/component/support/Function1.java
package com.xiaojinzi.component.support;

import android.support.annotation.NonNull;

import com.xiaojinzi.component.anno.support.CheckClassNameAnno;

@CheckClassNameAnno
public interface Function1<T, R> {

    /**
     * 做一个转化,从一个对象变成另一个对象
     */
    @NonNull
    R apply(@NonNull T t);

}

File: ComponentImpl/src/main/java/com/xiaojinzi/component/support/Function1.kt
package com.xiaojinzi.component.support

import com.xiaojinzi.component.anno.support.CheckClassNameAnno

@CheckClassNameAnno
interface Function1<T, R> {

    /**
     * 做一个转化,从一个对象变成另一个对象
     */
    fun apply(t: T): R

}